control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -83,20 +86,54 @@ func (r *MeshGatewayController) SetupWithManager(mgr ctrl.Manager) error {
 //  4. Role
 //  5. RoleBinding
 func (r *MeshGatewayController) onCreateUpdate(ctx context.Context, req ctrl.Request, resource *meshv2beta1.MeshGateway) error {
-	builder := gateways.NewMeshGatewayBuilder(resource)
+	//fetch gatewayclassconfig
+	gcc, err := r.getGatewayClassConfigForGateway(ctx, resource)
+	if err != nil {
+		return err
+	}
+
+	builder := gateways.NewMeshGatewayBuilder(resource, r.GatewayConfig, gcc)
 
 	upsertOp := func(ctx context.Context, _, object client.Object) error {
 		_, err := controllerutil.CreateOrUpdate(ctx, r.Client, object, func() error { return nil })
 		return err
 	}
 
-	err := r.opIfNewOrOwned(ctx, resource, &corev1.ServiceAccount{}, builder.ServiceAccount(), upsertOp)
+	err = r.opIfNewOrOwned(ctx, resource, &corev1.ServiceAccount{}, builder.ServiceAccount(), upsertOp)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to create service account: %w", err)
 	}
 
 	// TODO NET-6392 NET-6393 NET-6395
 
+	//Create deployment
+
+	mergeDeploymentOp := func(ctx context.Context, existingObject, object client.Object) error {
+		existingDeployment, ok := existingObject.(*appsv1.Deployment)
+		if !ok && existingDeployment != nil {
+			return fmt.Errorf("unable to infer existingDeployment type")
+		}
+		builtDeployment, ok := object.(*appsv1.Deployment)
+		if !ok {
+			return fmt.Errorf("unable to infer built deployment type")
```

Comment:
NIT: this casing is different from line 114, `existingDeployment` vs `built deployment`